app/public/consoleWindow.js

```diff
@@ -0,0 +1,92 @@
+const consoleWindowElement = document.getElementById("consoleWindow");
+
+const lineHolderElement = document.createElement("div");
+lineHolderElement.className = "lineHolder";
+consoleWindowElement.appendChild(lineHolderElement);
+
+const scrollElement = consoleWindowElement;
+
+const snapToBottomTolerance = 50; //When within this many pixels from the bottom of the console, new messages will cause the scroll to snap to the bottom
+let modes = ["public", "private", "important", "critical"];
+let seconds = 60*4; //for testing
+
+//Expects an object with 3 keys: time, message, classString
+function AddConsoleLine(lineObj){
+    let {time, message, classString} = lineObj;
+
+    let stringMessage = "";
+
+    if(time !== undefined){
+
+        let timeString = new Date(time * 1000).toISOString().substring(14, 19);
+        stringMessage += `${timeString} `;
+    }
+
+    stringMessage += "> "+message;
+    const newElem = AddConsoleLineString(stringMessage);
+    newElem.className = classString;
+}
+
+//Makes and returns a line element as a child of the lines parent element, setting its text content
+function AddConsoleLineString(lineString){
+    let scrolledToBottom = IsScrolledToBottom();
+    let lineElement = document.createElement("div");
+    lineElement.textContent = lineString;
+    lineHolderElement.appendChild(lineElement);
+
+    if(scrolledToBottom){
+        setTimeout(ScrollToBottom, 0); //delay allows the font time to adjust
+    }
+    return lineElement;
+}
+
+function RemoveChildrenOfElement(element){
+    while(element.firstChild){
```

Comment:
to remove child elements of an element, you can just. 
```
element.innerText = '';
```
This is so we don't need to iterate the element we can just set the property to empty string and garbage collection will probably clear the elements in the background